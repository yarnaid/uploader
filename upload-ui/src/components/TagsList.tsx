import React from "react";
import Tag from "./Tag";
import Stack from "react-bootstrap/Stack";

interface TagsListProps {
  tags: { value: string; checked: boolean }[];
  name: string;
}

const TagsList: React.FC<TagsListProps> = ({ tags, name }) => {
  if (!tags) return null;

  return (
    <Stack direction="horizontal" gap={2}>
      {tags.map((tag) => (
        <Tag
          key={tag.value}
          value={tag.value}
          name={name}
          checked={tag.checked}
        />
      ))}
    </Stack>
  );
};

export default TagsList;
