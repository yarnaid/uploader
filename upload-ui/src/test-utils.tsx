import React, { ReactElement } from "react";
import { Provider } from "react-redux";
import { render, RenderOptions } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import configureMockStore from "redux-mock-store";
import { configureStore } from "@reduxjs/toolkit";
import rootReducer from "./redux/reducers";
import {
  FileMetadataType,
  FilterType,
  FileType,
  UploadFileType,
} from "./types";
import type { PreloadedState } from "@reduxjs/toolkit";
import type { RootState, AppStore } from "./redux/store";

// This type interface extends the default options for render from RTL, as well
// as allows the user to specify other things such as initialState, store.
interface ExtendedRenderOptions extends Omit<RenderOptions, "queries"> {
  preloadedState?: PreloadedState<RootState>;
  store?: AppStore;
}

const mockStore = configureMockStore();

const initialState: {
  files: { files: FileType[] };
  filters: FilterType[];
  file: string;
  tags: FilterType[];
  metadata: FileMetadataType;
} = {
  files: { files: [] },
  filters: [],
  file: "",
  tags: [],
  metadata: { tags: [] },
};

export const store: any = mockStore(initialState);

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <Provider store={store}>
      <MemoryRouter>{children}</MemoryRouter>
    </Provider>
  );
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = configureStore({ reducer: rootReducer, preloadedState }),
    ...renderOptions
  }: ExtendedRenderOptions = {}
) => render(ui, { wrapper: AllTheProviders, ...options });
export * from "@testing-library/react";

export { customRender as render };
