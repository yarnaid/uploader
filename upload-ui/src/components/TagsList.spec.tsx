import React from "react";
import { render } from "../test-utils";
import { screen } from "@testing-library/react";
import TagsList from "./TagsList";

describe("TagsList", () => {
  const value = "test tag1";
  const value2 = "test tag2";
  const tags = [
    { value: value, checked: true },
    { value: value2, checked: false },
  ];
  it("renders 2 tags", () => {
    const { asFragment } = render(<TagsList tags={tags} name="test" />);
    expect(screen.getByText(/test tag2/i)).toBeInTheDocument();
    expect(screen.getByText(/test tag2/i)).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
