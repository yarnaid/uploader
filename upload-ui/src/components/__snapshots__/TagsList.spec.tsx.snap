// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TagsList > renders 2 tags 1`] = `
<DocumentFragment>
  <div
    class="hstack gap-2"
  >
    <span
      class="badge rounded-pill bg-primary"
    >
      test tag1
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M14 1a1 1 0 0 1 1 1v12a1 1 0 0 1-1 1H2a1 1 0 0 1-1-1V2a1 1 0 0 1 1-1h12zM2 0a2 2 0 0 0-2 2v12a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2H2z"
        />
        <path
          d="M4.646 4.646a.5.5 0 0 1 .708 0L8 7.293l2.646-2.647a.5.5 0 0 1 .708.708L8.707 8l2.647 2.646a.5.5 0 0 1-.708.708L8 8.707l-2.646 2.647a.5.5 0 0 1-.708-.708L7.293 8 4.646 5.354a.5.5 0 0 1 0-.708z"
        />
      </svg>
    </span>
    <span
      class="badge rounded-pill bg-secondary"
    >
      test tag2
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M14 1a1 1 0 0 1 1 1v12a1 1 0 0 1-1 1H2a1 1 0 0 1-1-1V2a1 1 0 0 1 1-1h12zM2 0a2 2 0 0 0-2 2v12a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2H2z"
        />
        <path
          d="M8 4a.5.5 0 0 1 .5.5v3h3a.5.5 0 0 1 0 1h-3v3a.5.5 0 0 1-1 0v-3h-3a.5.5 0 0 1 0-1h3v-3A.5.5 0 0 1 8 4z"
        />
      </svg>
    </span>
  </div>
</DocumentFragment>
`;

exports[`TagsList renders 2 tags 1`] = `
<DocumentFragment>
  <div
    class="hstack gap-2"
  >
    <span
      class="badge rounded-pill bg-primary"
    >
      test tag1
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M14 1a1 1 0 0 1 1 1v12a1 1 0 0 1-1 1H2a1 1 0 0 1-1-1V2a1 1 0 0 1 1-1h12zM2 0a2 2 0 0 0-2 2v12a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2H2z"
        />
        <path
          d="M4.646 4.646a.5.5 0 0 1 .708 0L8 7.293l2.646-2.647a.5.5 0 0 1 .708.708L8.707 8l2.647 2.646a.5.5 0 0 1-.708.708L8 8.707l-2.646 2.647a.5.5 0 0 1-.708-.708L7.293 8 4.646 5.354a.5.5 0 0 1 0-.708z"
        />
      </svg>
    </span>
    <span
      class="badge rounded-pill bg-secondary"
    >
      test tag2
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M14 1a1 1 0 0 1 1 1v12a1 1 0 0 1-1 1H2a1 1 0 0 1-1-1V2a1 1 0 0 1 1-1h12zM2 0a2 2 0 0 0-2 2v12a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2H2z"
        />
        <path
          d="M8 4a.5.5 0 0 1 .5.5v3h3a.5.5 0 0 1 0 1h-3v3a.5.5 0 0 1-1 0v-3h-3a.5.5 0 0 1 0-1h3v-3A.5.5 0 0 1 8 4z"
        />
      </svg>
    </span>
  </div>
</DocumentFragment>
`;
